reference/api/keys.md

```diff
@@ -1,30 +1,264 @@
 # Keys
 
-Each instance of MeiliSearch has three keys: a master, a private, and a public. Each key has a given set of permissions on the API routes.
+The `keys` route allows you to create, manage, and delete API keys. In order to use these endpoints, the [master key must be set](/reference/features/configuration.md#master-key) and supplied in the header of the request.
 
-You must have the master key to access the `keys` route.
+[Learn more about managing keys and their rights](/reference/features/authentication.md).
 
-[More information about the keys and their rights](/reference/features/authentication.md).
+## API key object definition
 
-## Get keys
+#### `description`
+
+**Type:** string
+**Modifiable:** yes
+**Mandatory:** no
+
+A description for the key. `null` if empty.
+
+#### `key`
+
+**Type:** string
+**Modifiable:** no
+
+An alphanumeric key value generated by MeiliSearch on API key creation. Used for authorization when making calls to a protected MeiliSearch instance. 
+
+This value is also used as the `:key` path variable to [get](#get-one-key), [update](#update-a-key), or [delete](#delete-a-key) a specific key.
+
+**Example**:
+
+```bash
+curl \
+  -X POST 'http://localhost:7700/indexes/products/documents' \
+  -H 'Authorization: Bearer d0552b41536279a0ad88bd595327b96f01176a60c2243e906c52ac02375f9bc4'
+  --data-binary '{
+    "id": 1,
+    "title": "Product 1",
+    "description": "Our first product!"
+    "price": 42
+  }'
+```
+
+#### `actions`
+
+**Type:** array
+**Modifiable:** yes
+**Mandatory:** yes
+
+A list of API actions permitted for the key. `["*"]` for all actions.
+
+The `*` character can also be used as a wildcard to get all permissions for a route, e.g. `documents.*` to authorize access to all document-related endpoints.
+
+| name             | description |
+|------------------|-------------|
+| search           | Provides access to `GET` and `POST` methods on `/indexes/:authorizedIndexes/search` routes. |
+| documents.add    | Provides access to `POST` and `PUT` on `/indexes/:authorizedIndexes/documents` routes. |
+| documents.get    | Provides access to `GET` methods on `/indexes/:authorizedIndexes/documents` and `/indexes/:authorizedIndexes/documents/:documentId` routes. |
+| documents.delete | Provides access to `DELETE` method on `/indexes/:authorizedIndexes/documents/:documentId`, `indexes/:authorizedIndexes/documents/:documentId` and `POST` method on `/indexes/:authorizedIndexes/documents/delete-batch` routes. |
+| indexes.add      | Provides access to `POST` `/indexes`. **⚠️ The newly created index is automatically added to the permissions for the API key making the call, as well as all others keys having `[*]` as a value for `indexes`**. |
+| indexes.get      | Provides access to `GET` `/indexes` and `/indexes/:authorizedIndexes`. **⚠️Non-authorized `indexes` are omitted from the response on `/indexes`**. |
```

Comment:
Do we really need the `⚠️  etc` bits in the description? They feel a bit more like warnings for the devs than for documentation users.